qml/DomainGraphLayout.qml

```diff
@@ -0,0 +1,1462 @@
+                            : max_host_width_
+
+                    // background
+                    Rectangle
+                    {
+                        id: host_background
+                        height: parent.height
+                        width: parent.width
+                        color: host_color_
+                        radius: radius_
+                    }
+
+                    // host name and icons
+                    Rectangle
+                    {
+                        id: host_tag
+                        anchors.horizontalCenter: parent.horizontalCenter
+                        implicitWidth: hostRowLayout.implicitWidth > max_host_width_
+                            ? hostRowLayout.implicitWidth
+                            : max_host_width_ == 0
+                                ? hostRowLayout.implicitWidth
+                                : max_host_width_
+                        height: label_height_
+                        color: host_color_
+                        radius: radius_
+
+                        RowLayout {
+                            id: hostRowLayout
+                            spacing: spacing_icon_label_
+                            anchors.centerIn: parent
+
+                            Rectangle {
+                                color: "transparent"
+                                width: modelData["status"] != "ok"
+                                    ? first_indentation_ : 0
+                            }
+                            IconSVG {
+                                visible: modelData["status"] != "ok"
+                                name: "issues"
+                                color: "white"
+                                size: modelData["status"] != "ok"? icon_size_ : 0
+                            }
+                            Rectangle {
+                                color: "transparent"
+                                width: first_indentation_ /2
+                            }
+                            IconSVG {
+                                name: "host"
+                                color: "white"
+                                size: icon_size_
+                            }
+                            Label {
+                                text: modelData["alias"]
+                                Layout.rightMargin: first_indentation_
+                                color: "white"
+                            }
+                        }
+                        Rectangle {
+                            visible: host_tag.implicitWidth < max_host_width_
+                            anchors.left: host_tag.right
+                            anchors.verticalCenter: parent.verticalCenter
+                            height: host_tag.height
+                            width: max_host_width_ - host_tag.implicitWidth
+                            color: host_background.color
+                            radius: radius_
+                        }
```

Comment:
Filling rectangle not needed? Delegated item already sets correct width if implicitWidth < max_width_. This applies to all sublisted items.